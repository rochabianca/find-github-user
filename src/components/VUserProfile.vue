<template>
  <div class="user-profile">
    <img
      data-test="user-avatar"
      class="user-profile__avatar"
      :src="user.avatar_url"
      alt="User profile"
    />
    <div data-test="user-name" class="user-profile__name">{{ user.name }}</div>
    <div data-test="user-bio" class="user-profile__bio">{{ user.bio }}</div>
  </div>
</template>

<script>
export default {
  name: 'UserProfile',
  props: {
    user: {
      type: Object,
      required: true,
      default: () => ({})
    }
  }
}
</script>

<style>
</style>
