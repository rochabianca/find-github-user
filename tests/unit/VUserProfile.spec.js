import { shallowMount } from '@vue/test-utils';
import VUserProfile from '@/components/VUserProfile';
import user from './fixtures/user';

describe('VUserProfile', () => {
  let props;

  const build = () => {
    const wrapper = shallowMount(VUserProfile, {
      propsData: props
    });
    return {
      wrapper,
      avatar: () => wrapper.find('[data-test="user-avatar"]'),
      name: () => wrapper.find('[data-test="user-name"]'),
      bio: () => wrapper.find('[data-test="user-bio"]')
    };
  };

  beforeEach(() => {
    props = {
      user
    };
  });

  it('renders the component', () => {
    const { wrapper } = build();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders main components', () => {
    const { avatar, name, bio } = build();

    expect(avatar().exists()).toBe(true);
    expect(avatar().attributes().src).toBe(props.user.avatar_url);

    expect(name().exists()).toBe(true);
    expect(name().text()).toBe(props.user.name);

    expect(bio().exists()).toBe(true);
    expect(bio().text()).toBe(props.user.bio);
  });
});
