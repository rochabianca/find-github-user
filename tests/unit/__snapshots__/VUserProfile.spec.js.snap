// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VUserProfile renders the component 1`] = `
<div class="user-profile"><img data-test="user-avatar" src="https://github.com/images/error/octocat_happy.gif" alt="User profile" class="user-profile__avatar">
  <div data-test="user-name" class="user-profile__name">monalisa octocat</div>
  <div data-test="user-bio" class="user-profile__bio">There once was...</div>
</div>
`;
